registry/zookeeper/registry.go

```diff
@@ -129,12 +129,17 @@ func (r *zkRegistry) InitListeners() {
 		recoverd := r.dataListener.subscribed
 		if recoverd != nil && len(recoverd) > 0 {
 			// recover all subscribed url
-			for conf, oldListener := range recoverd {
+			for recoveredURL, oldListener := range recoverd {
 				if regConfigListener, ok := oldListener.(*RegistryConfigurationListener); ok {
 					regConfigListener.Close()
 				}
-				newDataListener.SubscribeURL(conf, NewRegistryConfigurationListener(r.client, r))
-				go r.listener.ListenServiceEvent(conf, fmt.Sprintf("/dubbo/%s/"+constant.DEFAULT_CATEGORY, url.QueryEscape(conf.Service())), newDataListener)
+				serviceURL, err := common.NewURL(recoveredURL)
```

Comment:
There will use key of subscribed map to recover , so i think it can not change to service key.

@hxmhlt what do you think?